core/src/main/java/overflowdb/storage/NodeSerializer.java

```diff
@@ -17,38 +17,41 @@
 import java.util.function.Function;
 
 public class NodeSerializer extends BookKeeper {
-  private final OdbStorage storage;
-  private final Function<Object, Object> convertPropertyForPersistence;
+    private final OdbStorage storage;
+    private final Function<Object, Object> convertPropertyForPersistence;
+    private final ThreadLocal<MessageBufferPacker> packerPool = ThreadLocal.withInitial(MessagePack::newDefaultBufferPacker);
 
-  public NodeSerializer(boolean statsEnabled, OdbStorage storage, Function<Object, Object> convertPropertyForPersistence) {
-    super(statsEnabled);
-    this.storage = storage;
-    this.convertPropertyForPersistence = convertPropertyForPersistence;
-  }
+    public NodeSerializer(boolean statsEnabled, OdbStorage storage, Function<Object, Object> convertPropertyForPersistence) {
+        super(statsEnabled);
+        this.storage = storage;
+        this.convertPropertyForPersistence = convertPropertyForPersistence;
+    }
 
-  public NodeSerializer(boolean statsEnabled, OdbStorage storage) {
-    this(statsEnabled, storage, null);
-  }
+    public NodeSerializer(boolean statsEnabled, OdbStorage storage) {
+        this(statsEnabled, storage, null);
+    }
 
-  public byte[] serialize(NodeDb node) throws IOException {
-    long startTimeNanos = getStartTimeNanos();
-    try (MessageBufferPacker packer = MessagePack.newDefaultBufferPacker()) {
-      NodeLayoutInformation layoutInformation = node.layoutInformation();
-      /* marking as clean *before* we start serializing - if node is modified any time afterwards it'll be marked as dirty */
-      node.markAsClean();
+    public byte[] serialize(NodeDb node) throws IOException {
+        long startTimeNanos = getStartTimeNanos();
+        MessageBufferPacker packer = packerPool.get();
```

Comment:
Instead of `MessagePack.newDefaultBufferPacker` for every serialize operation, we reuse the existing one from the pool.